docs-src/cloud/limits.md

```diff
@@ -169,7 +169,42 @@ These operations create Events that are then logged in the Event History.
 The transaction size limit restricts the total size of Events that can be accommodated within a single transaction.
 
 The size limit for any given [Event History](/concepts/what-is-an-event-history) transaction is 4 MB.
-This isn't configurable.
+This limit is non-configurable for Temporal Cloud.
+
+**What is the blob size limit for a transaction?**
+
+- **Blob size limit** for Payloads, including Workflow context and each Workflow and Activity argument and return value:
+  - Temporal warns at 256 KB: `Blob size exceeds limit`.
+  - Temporal errors at 2 MB: `ErrBlobSizeExceedsLimit: Blob data size exceeds limit.`
+
+This limit is non-configurable for Temporal Cloud.
+
+The following provides solutions for handling large payloads.
+
+**Solutions for handling large payloads**
+
+If you're at risk of reaching the blob size limit, consider the following solutions:
+
+1. Use compression with a [custom Data Converter](/concepts/what-is-a-custom-data-converter) for large payloads.
+2. Break larger batches of commands into smaller batch sizes.
```

Comment:
Might want to give explanation on how to "break batches of commands". That might not be clear for users. This may be done either at the workflow-level (i.e., have the workflow itself handle smaller batches), or at the workflow _task_ level (i.e., wait on a 1ms timer after each batch of X scheduled activities).